iceoryx_posh/include/iceoryx_posh/internal/mepoo/mepoo_segment.inl

```diff
@@ -72,7 +73,23 @@ inline SharedMemoryObjectType MePooSegment<SharedMemoryObjectType, MemoryManager
 {
     return std::move(
         typename SharedMemoryObjectType::Builder()
-            .name(writerGroup.getName())
+            .name([&writerGroup] {
+                using ShmName_t = detail::PosixSharedMemory::Name_t;
+                ShmName_t shmName = iceoryxResourcePrefix(roudi::DEFAULT_UNIQUE_ROUDI_ID,
+                                                          "p"); // use an additional 'p' to prevent creating a payload
+                                                                // segment with the same name as the management segment
+                if (shmName.size() + writerGroup.getName().size() > ShmName_t::capacity())
+                {
+                    IOX_LOG(FATAL,
+                            "The payload segment with the name '"
+                                << writerGroup.getName().size()
+                                << "' would exceed the maximum allowed size when used with the '" << shmName
+                                << "' prefix!");
+                    IOX_PANIC("The shm name exceeds the max size with the 'iox1_#_p_' prefix");
```

Comment:
Why not use `shmName` here directly in the error message. Is it the same `p` btw?

Reply:
`IOX_PANIC` currently does not support logging like behavior. One can only use a string message.

Reply:
(optional) Then it is a topic for our later selves. Our, could we use the fatal log level and then call panic with an empty message?

Reply:
Yes, a panic with an empty message would be possible.